### PR TITLE
Add new keyboard: Commodore Amiga 1200 Membrane Converter

### PR DESCRIPTION
**Pull request by 8bits4ever**
## Description

Modification of the Model-M 101 membrane adapter (converter/modelm101) to work with Amiga 1200 (and possibly Amiga 500/600). A small PCB adapter needs to be used to make the necessary connections between the membrane and the Teensy 2.0 board.

This is meant for using the original Amiga keyboard as a USB input device, either with a desktop PC or a R-Pi/FPGA board hosted inside the Amiga case.

Info.json, and keymap files have been modified to fit Amiga keyboard layout

### CODE REVIEW

File: keyboards/converter/A1200/A1200.c [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+void led_set_kb(uint8_t usb_led) {
+  if (usb_led & (1<<USB_LED_CAPS_LOCK)) {
+    writePinLow(B6);
+  } else {
+    writePinHigh(B6);
+  }
+  
+
+  led_set_user(usb_led);
+}
```
Comment (drashna) [inline keyboards/converter/A1200/A1200.c:54]: The empty functions should be removed, and the LED can be set by a define. 
```suggestion
```

File: keyboards/converter/A1200/config.h [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+/*
+ * Feature disable options
+ *  These options are also useful to firmware size reduction.
+ */
+
+/* disable debug print */
+//#define NO_DEBUG
+
+/* disable print */
+//#define NO_PRINT
```
Comment (drashna) [inline keyboards/converter/A1200/config.h:115]:
```suggestion
```

File: keyboards/converter/A1200/keymaps/8bits4ever/config.h [hunk 3]
```diff
@@ -0,0 +1,19 @@
+/* Copyright 2021 8bits4ever
```
Comment (drashna) [inline keyboards/converter/A1200/keymaps/8bits4ever/config.h:1]: This file can be removed.

File: keyboards/converter/A1200/keymaps/8bits4ever/keymap.c [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+}
+
+void matrix_scan_user(void) {
+
+}
+
+void led_set_user(uint8_t usb_led) {
+
+}
```
Comment (drashna) [inline keyboards/converter/A1200/keymaps/8bits4ever/keymap.c:55]:
```suggestion
```

File: keyboards/converter/A1200/readme.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+Pins:        B6       5V    
+```  
+
+Please note that some versions of Model M have LED connected with a ribbon cable, and not with separate wires.  
+
+Keyboard Maintainer: [8bits4ever](https://github.com/8bits4ever)  
+Hardware Supported: Teensy 2.0++ board by PJRC  
+Hardware Availability: https://www.pjrc.com/store/teensypp.html 
+
+Make example for this keyboard (after setting up your build environment):
```
Comment (drashna) [inline keyboards/converter/A1200/readme.md:28]: could you add instructions on how to enter the bootloader?

File: keyboards/converter/A1200/rules.mk [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+MCU = at90usb1286
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
```
Comment (drashna) [inline keyboards/converter/A1200/rules.mk:11]:
```suggestion
```

File: keyboards/converter/A1200/rules.mk [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+BOOTLOADER = halfkay
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
```
Comment (drashna) [inline keyboards/converter/A1200/rules.mk:17]:
```suggestion
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
```

File: keyboards/converter/A1200/rules.mk [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+MOUSEKEY_ENABLE = no        # Mouse keys
+EXTRAKEY_ENABLE = no        # Audio control and System control
```
Comment (drashna) [inline keyboards/converter/A1200/rules.mk:19]:
```suggestion
EXTRAKEY_ENABLE = yes       # Audio control and System control
```

File: keyboards/converter/A1200/rules.mk [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality on B7 by default
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
```
Comment (drashna) [inline keyboards/converter/A1200/rules.mk:31]:
```suggestion
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality 
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
MIDI_ENABLE = no            # MIDI support
UNICODE_ENABLE = no         # Unicode
BLUETOOTH_ENABLE = no       # Enable Bluetooth 
AUDIO_ENABLE = no           # Audio output
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/converter/A1200/keymaps/8bits4ever/config.h (hunk 3 above)
Comment (8bits4ever) [reply keyboards/converter/A1200/keymaps/8bits4ever/config.h]: File has been removed

File: keyboards/converter/A1200/readme.md (hunk 5 above)
Comment (8bits4ever) [reply keyboards/converter/A1200/readme.md:30]: Added a small instructions. The Teensy board has a reset button for programming, after the board is flashed you can use a keyboard shortcut for the same purpose

File: keyboards/converter/A1200/A1200.c (hunk 1 above)
Comment (8bits4ever) [reply keyboards/converter/A1200/A1200.c:54]: Im new to the firmware. How do you set the caps lock LED by a define?

**Review by drashna — CHANGES_REQUESTED** [review]
Could you rename all of the files to be lower cased?

**Comment by 8bits4ever** [comment]
> Could you rename all of the files to be lower cased?

Done. Renamed A1200.c and A1200.h with lower case.

**Review by drashna — APPROVED** [review]

File: keyboards/converter/a1200/rules.mk [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality 
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+UNICODE_ENABLE = no         # Unicode
```
Comment (fauxpark) [inline keyboards/converter/a1200/rules.mk:22]:
```suggestion
```

File: keyboards/converter/a1200/readme.md [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+--------------------------------------------------------  
+Status LEDs: CapsLock +5V   
+Pins:        B6       5V    
+```  
+
+Please note that some versions of Model M have LED connected with a ribbon cable, and not with separate wires.  
+
+Keyboard Maintainer: [8bits4ever](https://github.com/8bits4ever)  
+Hardware Supported: Teensy 2.0++ board by PJRC  
+Hardware Availability: https://www.pjrc.com/store/teensypp.html
```
Comment (fauxpark) [inline keyboards/converter/a1200/readme.md:26]:
```suggestion
* Keyboard Maintainer: [8bits4ever](https://github.com/8bits4ever)
* Hardware Supported: Teensy 2.0++ board by PJRC
* Hardware Availability: https://www.pjrc.com/store/teensypp.html
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/converter/a1200/readme.md [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+
+* Keyboard Maintainer: [8bits4ever](https://github.com/8bits4ever)
+* Hardware Supported: Teensy 2.0++ board by PJRC
+* Hardware Availability: https://www.pjrc.com/store/teensypp.html
+
+For first time flashing use the reset button on the Teensy board enter in bootloader mode. Once the board is flashed you can use "HELP"+"R" on the Amiga keyboard for the same purpose.  
+
+Make example for this keyboard (after setting up your build environment):
+
+    make converter/A1200:8bits4ever
```
Comment (fauxpark) [inline keyboards/converter/a1200/readme.md:32]:
```suggestion
    make converter/a1200:default
```
This board also needs a `default` keymap.

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**